update_doc.py

```diff
@@ -122,15 +126,155 @@ def check_supported_models(check: bool):
             f.write(final_doc)
 
 
-def main():
-    parser = argparse.ArgumentParser()
-    parser.add_argument("--check", action="store_true")
+def start_server_and_wait():
+    log_file = open("/tmp/server_log.txt", "w")
 
-    args = parser.parse_args()
+    process = subprocess.Popen(
+        ["text-generation-launcher"],
+        stdout=log_file,
+        stderr=subprocess.STDOUT,
+        universal_newlines=True,
+    )
+    print("Server is starting...")
+
+    start_time = time.time()
+    while True:
+        try:
+            response = requests.get("http://127.0.0.1:3000/health")
+            if response.status_code == 200:
+                print("Server is up and running!")
+                return process, log_file
+        except requests.RequestException:
+            # timeout after 3 minutes (CI can be slow sometimes)
+            if time.time() - start_time > 180:
+                log_file.close()
+                with open("/tmp/server_log.txt", "r") as f:
+                    print("Server log:")
+                    print(f.read())
+                os.remove("/tmp/server_log.txt")
+                raise TimeoutError("Server didn't start within 60 seconds")
+            time.sleep(1)
+
+
+def stop_server(process, log_file, show=False):
+    process.terminate()
+    process.wait()
+    log_file.close()
+
+    if show:
+        with open("/tmp/server_log.txt", "r") as f:
+            print("Server log:")
+            print(f.read())
+    os.remove("/tmp/server_log.txt")
+
+
+def get_openapi_json():
+    response = requests.get("http://127.0.0.1:3000/api-doc/openapi.json")
+    # error if not 200
+    response.raise_for_status()
+    return response.json()
+
+
+def update_openapi_json(new_data, filename="docs/openapi.json"):
+    with open(filename, "w") as f:
+        json.dump(new_data, f, indent=2)
+
+
+def compare_openapi(old_data, new_data):
```

Comment:
Honestly I wouldn't even ever parse the files and simply reuse some simple diffing.

```
diff = subprocess.run(
                    ["diff", tmp, filename], capture_output=True
                ).stdout.decode("utf-8")
print(diff)
```

Not strong opinion, your code seems cleaner, but given the unreadableness of the original JSON file, I'm not sure it's worth potentially making mistakes in that diff algorithm.

Reply:
good points! I think `diff` is a better long term solution and the pr has been updated to use `diff`